cpp/dolfinx/fem/FiniteElement.h

```diff
@@ -691,7 +693,11 @@ class FiniteElement
       _sub_elements;
 
   // Dimension of each value space
-  std::vector<std::size_t> _value_shape;
+  std::vector<std::size_t> _reference_value_shape;
+  std::vector<std::size_t> _value_shape_0;
+  std::vector<std::size_t> _value_shape_1;
+  std::vector<std::size_t> _value_shape_2;
+  std::vector<std::size_t> _value_shape_3;
```

Comment:
```suggestion
  std::array<std::vector<std::size_t>, 4> _value_shape;
```
I think this looks a lot better as an array (or vector) of vectors.

Reply:
resolving as I've removed these variables now